tests/plan_lifecycle/test_plan_archive_pvc_cleanup.py

```
@@ -0,0 +1,330 @@
+                partial_name=session_uuid,
+            ):
+                if not sample:
+                    return
+        except TimeoutExpiredError:
+            orphan_names = get_orphan_resource_names(
+                client=ocp_admin_client, namespace=vm_namespace, partial_name=session_uuid
+            )
+            raise AssertionError(
+                f"Orphan resources with session '{session_uuid}' remain in "
+                f"namespace '{vm_namespace}' after plan archive+delete: {orphan_names}"
+            )
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)